app/controllers/discussions_messages_controller.rb

```diff
@@ -11,7 +11,10 @@ def create
   end
 
   def destroy
-    current_message.destroy!
+    authorize_moderator! unless params[:motive] == :self_deleted.to_s
+
+    current_message.soft_delete! params[:motive], current_user
+    current_message.update! not_actually_a_question: true
```

Comment:
I get why this was introduced, but it seems a little brittle.
I think the best solution would be to properly scope discussion messages so that they don't include deleted messages at all in `update_counters!` method of `discussion`. What do you think?

Reply:
Yes, did it on a later commit.